### ISSUE TITLE
Txpool not receiving any new pending transactions

### ISSUE THREAD

**Issue by revofever** [ethereum/go-ethereum#21320]
Originally posted on stackexchange: https://ethereum.stackexchange.com/questions/84815/txpool-not-receiving-any-new-pending-transactions

#### System information

Geth version: `1.9.15-stable`
OS & Version: Ubuntu 18.04.3 LTS
Digital Ocean Droplet
6 vCPUs
16 GB RAM
#### Expected behaviour

Transactions being added to the txpool at all times, including when there is an influx of new transactions and when the txpool count gets large.

The following subscription should stay operational and stable:

```
web3.eth.subscribe('pendingTransactions', function(error, result){
    if (!error)
        console.log(result);
})
.on("data", function(transaction){
    console.log(transaction);
    let promise = web3.eth.getTransaction(transaction);
    promise.then((data) => {
	    if(data){
		    console.log(data);
	    }
    });
});
```

#### Actual behaviour

It seems the node will stop receiving new transactions into the txpool at times of high activity, or even randomly.

Calling txpool.status
`{
  pending: 144598,
  queued: 7244
}`

5 mins later: `{
  pending: 144598,
  queued: 7747
}
`

The pending number is not increasing, while transactions are still being added to queued. Transactions are not received from the web3 subscription at this time.

Everything will return to normal after a period of time (15 mins - x hours).

Memory usage is normal and stable (60%), CPU usage nowhere near maxed out.

#### Steps to reproduce the behaviour

Run a node with high txpool limits, and the web3 subscription listed above. Test to see when the txpool pending count stalls.
`--txpool.globalslots 1000000 --txpool.accountslots 128 --txpool.accountqueue 512 --txpool.globalqueue 500000 --txpool.lifetime 24h0m0s`

**Comment by rjl493456442** [comment]
Can you try the latest release? We have a fix for txpool which may or may not solve this issue.

**Comment by revofever** [comment]
I will update if the issue continues on the latest release. It is rare so it could take some time.

**Comment by revofever** [comment]
The issue disappeared after I did a few things. First updated to the latest version and set a lower txpool limit. Now I am using 75000 max `globalslots`.

Next I wrote a script that will run every minute. It drops nodes that are way behind in syncing. I did see a small improvement doing this, but I'm not sure if it was the main cause of the issue. I wouldn't recommend doing this unless you are using the node as a service provider. In my case, I need as close to 100% uptime as I can get.
